### PR TITLE
Add attribute for original filepath to Component.  Addresses DDR-909.

### PR DESCRIPTION
**Pull request by coblej**
Also makes the attribute available on the Item if the Item has one Component.

### CODE REVIEW

File: lib/ddr/models/solr_document.rb [hunk 1]
```diff
@@ -11,19 +11,20 @@ module SolrDocument
       alias_method :pid, :id
     end
 
-    class NotFound < Error; end
+    class NotFound < Error;
+    end
```
Comment (dchandekstark) [inline lib/ddr/models/solr_document.rb]: Can we take out the stylistic changes in SolrDocument?  I'd like to have a separate conversation about those. The one above is just one example.

File: lib/ddr/models/solr_document.rb [hunk 2]
```diff
@@ -113,7 +115,7 @@ def admin_policy
     end
 
     def has_children?
-      ActiveFedora::SolrService.class_from_solr_document(self).reflect_on_association(:children).present?
+      children.present?
```
Comment (dchandekstark) [inline lib/ddr/models/solr_document.rb]: Is this change germane to the essential change in the PR? Not sure I object, but please explain.

File: lib/ddr/models/solr_document.rb [hunk 3]
```diff
@@ -248,6 +252,24 @@ def rights
       self["rights_tesim"]
     end
 
+    def original_filepath
+      if self[Ddr::Index::Fields::ORIGINAL_FILEPATH]
+        self[Ddr::Index::Fields::ORIGINAL_FILEPATH]
+      elsif self[Ddr::Index::Fields::ACTIVE_FEDORA_MODEL] == 'Item'
+        if childrn = children
```
Comment (dchandekstark) [inline lib/ddr/models/solr_document.rb]: I don't see the need for this assignment.

File: lib/ddr/models/solr_document.rb [hunk 4]
```diff
@@ -248,6 +252,24 @@ def rights
...
+            childrn.first.original_filepath
+          end
+        end
+      end
+    end
+
+    def children
+      if qry = children_query
+        qry.docs
+      end
```
Comment (dchandekstark) [inline lib/ddr/models/solr_document.rb]: You could also write 

```ruby
children_query.docs rescue nil
```

Comment (dchandekstark) [inline lib/ddr/models/solr_document.rb]: Or perhaps you want `[]` as the default?

File: lib/ddr/models/solr_document.rb (hunk 2 above)
Comment (coblej) [inline lib/ddr/models/solr_document.rb]: No, not germane.  Since this PR adds the "children" method, it was just an opportunity to simplify this code.  Can revert the change if you prefer.

File: lib/ddr/models/solr_document.rb (hunk 3 above)
Comment (coblej) [inline lib/ddr/models/solr_document.rb]: just so I don't call the "children" method multiple times, re-executing the query each time

File: lib/ddr/models/solr_document.rb (hunk 4 above)
Comment (coblej) [inline lib/ddr/models/solr_document.rb]: yeah, [] is probably a better default than nil

File: lib/ddr/models/solr_document.rb (hunk 2 above)
Comment (coblej) [inline lib/ddr/models/solr_document.rb]: Reverted the change.

**Comment by coblej** [comment]
@dchandekstark PR revised in response to comments.  Ready for review.
